### PR TITLE
fix: Added missing JS toggles

### LINKED ISSUE #16743: [Bug]: Missing JS toggles in widgets and JSON Form columns
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The Default Country Code property in the Phone Input field has a JS toggle. However, this is missing inside the JSON Form.
Screenshot 2022-09-14 at 11 04 19

### Steps To Reproduce

Create a custom column for Phone Input inside a JSON Form. Notice that the Country Code property does not have a JS toggle.

### Public Sample App

_No response_

### Version

v1.8.1

### PR DESCRIPTION
**Pull request by trishitapingolia**
## Description

Added JS toggles for:
- Phone input regular
-Allow country code change

Currency input regular

- Allow currency change
- Decimals allowed

Fixes #16743 

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

How did you test the change @trishitapingolia

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by somangshu** [comment]
@trishitapingolia can you please go through the checklist of your PR description while our reviewers look into this

File: app/client/src/widgets/CurrencyInputWidget/widget/index.tsx [hunk 1]
```diff
@@ -169,7 +169,8 @@ class CurrencyInputWidget extends BaseInputWidget<
                   value: 2,
                 },
               ],
-              isBindProperty: false,
+              isJSConvertible: true,
```
Comment (aswathkk) [inline app/client/src/widgets/CurrencyInputWidget/widget/index.tsx:172]: @trishitapingolia We need to add a validation here.

**Review by aswathkk — CHANGES_REQUESTED** [review]
@trishitapingolia We need to add a validation whenever a property is made JS convertible.

Also, there are some more properties that need to be made JS convertible. Refer [this comment](https://github.com/appsmithorg/appsmith/issues/16743#issuecomment-1246264850) on the same issue.

**Comment by trishitapingolia** [comment]
> @trishitapingolia We need to add a validation whenever a property is made JS convertible.
> 
> Also, there are some more properties that need to be made JS convertible. Refer [this comment](https://github.com/appsmithorg/appsmith/issues/16743#issuecomment-1246264850) on the same issue.

I have made the modifications, please review. I am having difficulty in other properties that need to be made js convertible. Can you review and merge this for now?

File: app/client/src/widgets/CurrencyInputWidget/widget/index.tsx [hunk 2]
```diff
@@ -169,10 +169,15 @@ class CurrencyInputWidget extends BaseInputWidget<
...
+              isBindProperty: true,
               isTriggerProperty: false,
+              validation: {
+              type: ValidationTypes.NUMBER,
```
Comment (aswathkk) [inline app/client/src/widgets/CurrencyInputWidget/widget/index.tsx:176]: @trishitapingolia The formatting is not proper here. Can you fix that?

**Review by aswathkk — CHANGES_REQUESTED** [review]
@trishitapingolia There's some issue with code formatting. Try using [prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) which will automatically format the code.

**Comment by trishitapingolia** [comment]
@aswathkk @somangshu can please someone review and merge this as it's the last day of Hacktoberfest.

**Comment by somangshu** [comment]
Hey @trishitapingolia please do not worry as this is last day of oct. We will still consider the PR and merge it if everything looks good.

Cc @aswathkk

**Comment by trishitapingolia** [comment]
> Hey @trishitapingolia please do not worry as this is last day of oct. We will still consider the PR and merge it if everything looks good.
> 
> Cc @aswathkk

Ok thanks

**Review by aswathkk — DISMISSED** [review]
LGTM

**Review by somangshu — APPROVED** [review]

**Pull request merged**